src/main/scala/sigmastate/utils/SigmaByteWriter.scala

```diff
@@ -1,14 +1,63 @@
 package sigmastate.utils
 
-import scorex.util.ByteArrayBuilder
-import scorex.util.serialization.VLQByteBufferWriter
+import scorex.util.serialization.{ByteStringWriter, VLQByteBufferWriter, Writer}
+import scorex.util.serialization.Writer.Aux
 import sigmastate.SType
 import sigmastate.Values.Value
 import sigmastate.serialization.{ConstantStore, TypeSerializer, ValueSerializer}
 
-class SigmaByteWriter(b: ByteArrayBuilder,
-                      val constantExtractionStore: Option[ConstantStore]) extends VLQByteBufferWriter(b) {
+// todo rename to SigmaWriter?
+class SigmaByteWriter(val w: Writer,
+                      val constantExtractionStore: Option[ConstantStore]) extends Writer {
 
-  @inline def putType[T <: SType](x: T): SigmaByteWriter = { TypeSerializer.serialize(x, this); this }
-  @inline def putValue[T <: SType](x: Value[T]): SigmaByteWriter = { ValueSerializer.serialize(x, this); this }
+  type CH = w.CH
+
+  override def length(): Int = w.length()
+
+  override def newWriter(): Aux[CH] = w.newWriter()
+
+  override def putChunk(chunk: CH): this.type = { w.putChunk(chunk); this }
+
+  override def result(): CH = w match {
+    case wr: ByteStringWriter => wr.result().asInstanceOf[CH]
```

Comment:
@greenhat Why not just always use `wr.result().asInstanceOf[CH]`?

Reply:
@ergomorphic Yep. I missed that. Thanks.